### PR TITLE
Fix dot prefix bug in DynamoCatalog

### PR DESCRIPTION
**Pull request by LingMQ**

### CODE REVIEW

File: aws/src/main/java/org/apache/iceberg/aws/dynamo/DynamoCatalog.java [hunk 1]
```diff
@@ -22,10 +22,20 @@ public DynamoCatalog(Configuration conf) {
...
   }
 
+  /**
+   * This method compose the identification string of a table.
+   * @param tableIdentifier the given table identifier
+   * @return the returned string will be "(namespace).(table name)". If there is no namespace, the returned string
+   * will be "(table name)"
+   */
+  private String getTableIdentifierStr(TableIdentifier tableIdentifier) {
+    return (tableIdentifier.namespace().isEmpty()) ? tableIdentifier.name() : tableIdentifier.toString();
```
Comment (eskrm) [inline aws/src/main/java/org/apache/iceberg/aws/dynamo/DynamoCatalog.java]: We should push this into `TableIdentifier.toString`. Also, you can use `tableIdentifier.hasNamespace` instead.

Comment (eskrm) [reply aws/src/main/java/org/apache/iceberg/aws/dynamo/DynamoCatalog.java]: Can you run all the tests after that change to make sure we didn't unintentionally break something?

Comment (LingMQ) [reply aws/src/main/java/org/apache/iceberg/aws/dynamo/DynamoCatalog.java]: Updated!

Comment (LingMQ) [reply aws/src/main/java/org/apache/iceberg/aws/dynamo/DynamoCatalog.java]: And the ./gradlew build ran through successfully

File: api/src/main/java/org/apache/iceberg/catalog/TableIdentifier.java [hunk 2]
```diff
@@ -77,6 +77,6 @@ public String name() {
   }
 
   public String toString() {
-    return namespace.toString() + "." + name;
+    return (namespace.isEmpty()) ? name : namespace.toString() + "." + name;
```
Comment (eskrm) [inline api/src/main/java/org/apache/iceberg/catalog/TableIdentifier.java]: nit: can you drop the "()" around `namespace.isEmpty()`

Comment (LingMQ) [reply api/src/main/java/org/apache/iceberg/catalog/TableIdentifier.java]: done~

**Pull request merged**